composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/presentation/CreateEventViewModel.kt

```diff
@@ -226,6 +252,63 @@ class CreateEventViewModel(dispatcherProvider: DispatcherProvider, savedStateHan
         }
     }
 
+    private fun onAttendeeEmailChange(text: String) {
+        if (viewStateFlow.value.isAttendeeEmailError) {
+            emitViewState { viewState ->
+                viewState.copy(
+                    isAttendeeEmailError = validateEmailUseCase(text)
+                )
+            }
+        }
+    }
+
+    private fun onInviteAttendee(email: String) {
+        val isEmailValid = validateEmailUseCase(email)
+        if (!isEmailValid) {
+            emitViewState { viewState ->
+                viewState.copy(isAttendeeEmailError = true)
+            }
+            return
+        }
+
+        viewModelScope.launch {
+            val attendeeResult = createEventRepository.getAttendee(email)
+
+            val attendee = attendeeResult.getOrNull()
+
+            if (attendeeResult.isFailure || attendee == null) {
+                emitViewState { viewState ->
+                    viewState.copy(
+                        isAttendeeEmailError = true
+                    )
+                }
+                return@launch
+            }
+
+            emitViewState { viewState ->
+                viewState.copy(
+                    attendees = mutableListOf<Attendee>().also {
+                        it.add(attendee)
+                        it.addAll(viewState.attendees)
+                    },
+                    showAttendeeDialog = false,
+                    isAttendeeEmailError = false
+                )
+            }
+        }
+    }
+
+    private fun onRemoveAttendee(attendee: Attendee) {
+        emitViewState { viewState ->
+            viewState.copy(
+                attendees = mutableListOf<Attendee>().also {
+                    viewState.attendees.filter { attendee.userId != it.userId }
+                }
```

Comment:
You probably tested this and it works, but in case not, I'm not sure if it does 🤔 
You convert the attendees list into a mutable one and then operate on that mutable list. This filtering without assigning a new instance to the list is something that's normally not detected as a state update by the compose compiler since you're not working with a `mutableStateListOf()`. I don't understand why you work with a mutable list here in the first place, you could just assign the filter to the field right away